cli/deployment/flags.go

```diff
@@ -375,17 +384,54 @@ func RemoveSensitiveValues(df codersdk.DeploymentFlags) codersdk.DeploymentFlags
 	t := v.Type()
 	for i := 0; i < t.NumField(); i++ {
 		fv := v.Field(i)
-		if v, ok := fv.Interface().(codersdk.StringFlag); ok {
-			if v.Secret && v.Value != "" {
+		if vp, ok := fv.Interface().(*codersdk.StringFlag); ok {
+			if vp.Secret && vp.Value != "" {
+				// Make a copy and remove the value.
+				v := *vp
 				v.Value = secretValue
-				fv.Set(reflect.ValueOf(v))
+				fv.Set(reflect.ValueOf(&v))
 			}
 		}
 	}
 
 	return df
 }
 
+//nolint:revive
+func AttachFlags(flagset *pflag.FlagSet, df *codersdk.DeploymentFlags, enterprise bool) {
+	v := reflect.ValueOf(df).Elem()
+	t := v.Type()
+	for i := 0; i < t.NumField(); i++ {
+		fv := v.Field(i)
+		fve := fv.Elem()
+		e := fve.FieldByName("Enterprise").Bool()
+		if e != enterprise {
+			continue
+		}
+		if e {
+			d := fve.FieldByName("Description").String()
+			d += cliui.Styles.Keyword.Render(" This is an Enterprise feature. Contact sales@coder.com for licensing")
+			fve.FieldByName("Description").SetString(d)
+		}
+
+		switch v := fv.Interface().(type) {
+		case *codersdk.StringFlag:
+			StringFlag(flagset, v)
+		case *codersdk.StringArrayFlag:
+			StringArrayFlag(flagset, v)
+		case *codersdk.IntFlag:
+			IntFlag(flagset, v)
+		case *codersdk.BoolFlag:
+			BoolFlag(flagset, v)
+		case *codersdk.DurationFlag:
+			DurationFlag(flagset, v)
```

Comment:
We should probably add a default case that panics or something here. `panic(fmt.Sprintf("unknown flag type %T", fv.Interface()))` should work